src/malli/core.cljc

```diff
@@ -2399,18 +2400,37 @@
   ([ns name ?schema data key f]
    (swap! -function-schemas* assoc-in [key ns name] (merge data {:schema (f ?schema), :ns ns, :name name}))))
 
+#?(:clj
+   (def resolve-fn
+     "Which function do we use for resolving var's namespaces?"
+     ;; If we're in a cljs env, we need to use the cljs analyzer
+     (if (find-ns 'cljs.analyzer.api)
+       (ns-resolve 'cljs.analyzer.api 'resolve)
+       resolve)))
+#?(:clj
+    (defn fix-syms
+      "Walk a data structure, resolve syms to their full names"
+      [env d]
+      (walk/postwalk
+       (fn [x]
+         (if (symbol? x)
+           (or (:name (resolve-fn env x)) x)
+           x))
+       d)))
+
 #?(:clj
    (defmacro => [name value]
      (let [name' `'~(symbol (str name))
            ns' `'~(symbol (str *ns*))
-           sym `'~(symbol (str *ns*) (str name))]
+           sym `'~(symbol (str *ns*) (str name))
+           value' (fix-syms &env value)]
```

Comment:
Because we're in the core ns here it's probably a good idea not create more publicly visible vars, how about wrapping the `defmacro` with a let?

some code golf, `(:ns &env)` is an idiom often used to determine if you're targeting a cljs macro:
```clojure
#?(:clj 
  (let [resolve' (if (:ns &env) (ns-resolve 'cljs.analyzer.api 'resolve) resolve)
      resolve-symbols
      (fn [env d]
        (walk/postwalk
         (fn [x]       
            (cond->> x (symbol? x)
             (or (:name (resolve' env x)))))
         d)))

value' (resolve-symbols &env value))
  (defmacro => ...
```

Reply:
I'm curious because you added the `resolve` for clojure schemas as well - do clojure schemas have this issue as well if you `instrument!` from another namespace?

Reply:
I'm... not actually sure. That's probably a good question to ask, it would simplify this whole mess a lot of I only need to call this in cljs